### ISSUE TITLE
Use variableLabels instead of constLabels

### ISSUE THREAD

**Issue by xanonid** [larscheid-schmitzhermes/prometheus-filesystem-exporter#1]
If there are two metrics with the same name, but different labels, prometheus-filesystem-exporter panics. This happens for example with [keycloak-monitoring-prometheus](https://github.com/larscheid-schmitzhermes/keycloak-monitoring-prometheus/issues) if more than on realm is active.

Example:
Run prometheus-filesystem-exporter and run

```bash
echo 1 > /metrics/answer_to_everything\;scope\=universe\;env\=prod
echo 1 > /metrics/answer_to_everything\;scope\=universe\;env\=dev
```
Result:
```go
2018/02/19 13:35:45 Attempting to generate metric from file: /metrics/answer_to_everything;scope=universe;env=prod
2018/02/19 13:35:45 Setting metric for /metrics/answer_to_everything;scope=universe;env=prod to 1.0000000000
2018/02/19 13:35:55 Attempting to update metric with value written to: /metrics/answer_to_everything;scope=universe;env=dev
2018/02/19 13:35:55 Attempting to generate metric from file: /metrics/answer_to_everything;scope=universe;env=dev
/ # panic: a previously registered descriptor with the same fully-qualified name as Desc{fqName: "answer_to_everything", help: "Auto generated from filesystem path: /metrics/answer_to_everything;scope=universe;env=dev", constLabels: {env="dev",scope="universe"}, variableLabels: []} has different label names or a different help string

goroutine 34 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc420044600, 0xc42006b9d0, 0x1, 0x1)
        /Users/tobiaslarscheid/go/src/github.com/prometheus/client_golang/prometheus/registry.go:353 +0x92
github.com/prometheus/client_golang/prometheus.MustRegister(0xc42006b9d0, 0x1, 0x1)
        /Users/tobiaslarscheid/go/src/github.com/prometheus/client_golang/prometheus/registry.go:152 +0x53
main.getOrCreateMetricForPath(0xc420128600, 0x34, 0x1, 0xc42006b960)
        /Users/tobiaslarscheid/go/src/github.com/larscheid-schmitzhermes/prometheus-filesystem-exporter/main.go:79 +0x33d
main.updateMetric(0xc420128600, 0x34)
        /Users/tobiaslarscheid/go/src/github.com/larscheid-schmitzhermes/prometheus-filesystem-exporter/main.go:89 +0x140
main.main.func1(0xc42011a0a0)
        /Users/tobiaslarscheid/go/src/github.com/larscheid-schmitzhermes/prometheus-filesystem-exporter/main.go:44 +0x13c
created by main.main
        /Users/tobiaslarscheid/go/src/github.com/larscheid-schmitzhermes/prometheus-filesystem-exporter/main.go:52 +0xee
```

**Comment by tobilarscheid** [comment]
Hi @xanonid,

thanks for reaching out - good catch! Do you think you could provide a PR in the respective repo?

**Comment by xanonid** [comment]
@tobilarscheid The fix itself would be more or less trivial, but already the unchanged code FTBFS:

```
./main.go:22: undefined: prometheus.Untyped
./main.go:64: undefined: prometheus.Untyped
./main.go:73: undefined: prometheus.NewUntyped
```

The reason is https://github.com/prometheus/client_golang/commit/6164fff8ce013e72b53d6b771c4fec72a80bbb0c . There is also a [comment](https://github.com/prometheus/client_golang/commit/6164fff8ce013e72b53d6b771c4fec72a80bbb0c#commitcomment-25457299) where it is described how this should be solved.

**Comment by tobilarscheid** [comment]
Hmmm, alright. So essentially instead of exposing untyped metrics we could move to gauges.

**Comment by tobilarscheid** [comment]
closing this as it is fixed by #2. Thank you once again!
